hack/stabilization-changes.py

```diff
@@ -28,29 +38,18 @@ def parse_iso8601_delay(delay):
     return datetime.timedelta(days=7*weeks, hours=hours)
 
 
-def stabilization_changes(directory):
-    channels = {}
-    paths = {}
-    for root, _, files in os.walk(directory):
-        for filename in files:
-            if not filename.endswith('.yaml'):
-                continue
-            path = os.path.join(root, filename)
-            with open(path) as f:
-                try:
-                    data = yaml.load(f, Loader=yaml.SafeLoader)
-                except ValueError as error:
-                    raise ValueError('failed to load YAML from {}: {}'.format(path, error))
-            channel = data['name']
-            if channel in channels:
-                raise ValueError('multiple definitions for {}: {} and {}'.format(channel, paths[channel], path))
-            paths[channel] = path
-            channels[channel] = data
+def stabilization_changes(directory, github_repo=None, github_token=None, webhook=None, waiting_notifications=False):
```

Comment:
Fair.  Suggestions on how to split it up?  It contains loops over channels, and loops over versions within each channel.  Both loop bodies could be split out into their own functions, but maybe that is overkill?

Reply:
Agree with David. Can we move each loop to separate functions?

Reply:
https://github.com/vrutkovs/cincinnati-graph-data-1/commit/83fa54db9ec142e1286da5e288a98ce75327a64c is my attempt to break it down in "fits on single screen" functions. Parameters are still a mess, we should be using objects to pass these effictiently

Reply:
I've borrow heavily from vrutkovs/cincinnati-graph-data-1@83fa54db9ec142e1286da5e288a98ce75327a64c in working up 981c51d -> 8a762a6, splitting up into some helper functions.  I didn't include `create_pull_request`, which wrapped `promote`, because it was mostly a string-creating and error-handling shim.  And while those take up some space, they also overlap somewhat with the "we don't actually want to promote this one yet" message, so I've left them inlined there one level up.